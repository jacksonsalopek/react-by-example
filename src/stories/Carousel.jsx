import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const CarouselButton = styled.button`
  background-color: rgba(0, 0, 0, 0.12);
  border: none;
  color: white !important;
  padding: 4px 12px;
  transition: all 0.5s;
  font-size: 16px;
  position: relative;
  display: block;
  ${props => props.id === 'prev' ? 'right: -80px;' : 'left: -80px;'}

  &:hover {
    background-color: rgba(0, 0, 0, 0.45);
  }
`

const CarouselEllipseButton = styled.button`
  background-color: ${props => props.active ? 'rgba(255, 255, 255, 0.45)' : 'rgba(0, 0, 0, 0.45)'};
  border-radius: 50%;
  display: block;
  width: 12px;
  height: 12px;
  border: none;
`

const CarouselImageWrapper = styled.div`
`

const CarouselImageButtonsContainer = styled.div`
  display: grid;
  grid-template-columns: ${(props) => `repeat(${props.imageCount}, 1fr)`};
  position: relative;
  top: -20px;
  justify-items: center;
  align-items: center;
  width: ${(props) => `${props.imageCount * 16}px`};
  margin: 0 auto;
`

const CarouselImage = styled.img`
  width: 100%;
  height: 100%;
`

const CarouselWrapper = styled.div`
  display: grid;
  grid-template-columns: 80px 1fr 80px;
  justify-content: center;
`

export const Carousel = ({ imageUrls }) => {
  const [currentImage, setCurrentImage] = React.useState(imageUrls[0])
  const [currentIndex, setCurrentIndex] = React.useState(0)
  const onClick = (type) => {
    if (type === 'next') {
      if (currentIndex === imageUrls.length - 1) {
        setCurrentIndex(0)
        setCurrentImage(imageUrls[0])
      } else {
        setCurrentIndex(currentIndex + 1)
        setCurrentImage(imageUrls[currentIndex + 1])
      }
    } else if (type === 'prev') {
      if (currentIndex === 0) {
        setCurrentIndex(imageUrls.length - 1)
        setCurrentImage(imageUrls[imageUrls.length - 1])
      } else {
        setCurrentIndex(currentIndex - 1)
        setCurrentImage(imageUrls[currentIndex - 1])
      }
    } else {
      setCurrentIndex(type)
      setCurrentImage(imageUrls[type])
    }
  }

  React.useEffect(() => {
    
  }, [])

  return (
    <CarouselWrapper>
      <CarouselButton onClick={() => onClick('prev')} id="prev" title="Previous image button">&lt;</CarouselButton>
      <CarouselImageWrapper>
        <CarouselImage src={currentImage} alt="Carousel image" />
        <CarouselImageButtonsContainer imageCount={imageUrls.length}>
          {imageUrls.map((imageUrl, index) => (
            <CarouselEllipseButton key={index} onClick={() => onClick(index)} active={index === currentIndex} />
          ))}
        </CarouselImageButtonsContainer>
      </CarouselImageWrapper>
      <CarouselButton onClick={() => onClick('next')} id="next" title="Previous image button">&gt;</CarouselButton>
    </CarouselWrapper>
  )
}

Carousel.propTypes = {
  imageUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
}

Carousel.defaultProps = {
  imageUrls: [
    'https://www.freecodecamp.org/news/content/images/2021/07/jackson-so-_t-l5FFH8VA-unsplash.jpg',
    'https://images.unsplash.com/photo-1681916882860-adfcc0a42d0e?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1332&q=80',
    'https://images.unsplash.com/photo-1682268329792-db6575d60b98?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1373&q=80'
  ],
}
