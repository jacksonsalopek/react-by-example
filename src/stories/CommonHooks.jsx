import React from 'react';
import PropTypes from 'prop-types'
import styled from 'styled-components'

const CommonHooksButton = styled.button`
  background-color: ${props => props.color ?? '#4CAF50'}; /* Green */
  border: none;
  color: white !important;
  padding: 4px 12px;
  border-radius: 4px;
  display: inline-block;
  margin: 4px 2px;
`

function reducer(state, action) {
  switch (action.type) {
    case 'ADD':
      return {
        ...state,
        items: [...state.items, action.payload]
      };
    case 'REMOVE':
      return {
        ...state,
        items: state.items.filter(item => item.id !== action.payload)
      };
    default:
      throw new Error();
  }
}

export const CommonHooks = ({ initialItems, ...props }) => {
  const [state, dispatch] = React.useReducer(reducer, { items: initialItems });

  const itemTotal = React.useMemo(() => {
    return state.items.reduce((acc, item) => acc + item.quantity, 0);
  }, [state.items]);

  function handleAddItem() {
    const newItem = { id: Date.now(), name: 'New Item', quantity: 1 };
    dispatch({ type: 'ADD', payload: newItem });
  }

  function handleRemoveItem(id) {
    dispatch({ type: 'REMOVE', payload: id });
  }

  return (
    <div data-testid="common-hooks">
      <div>Item Total: {itemTotal}</div>
      <CommonHooksButton onClick={handleAddItem}>Add Item</CommonHooksButton>
      <ul>
        {state.items.map(item => (
          <li key={item.id}>
            {item.name} - {item.quantity}
            <CommonHooksButton onClick={() => handleRemoveItem(item.id)} color={'red'}>Remove</CommonHooksButton>
          </li>
        ))}
      </ul>
    </div>
  );
}

CommonHooks.propTypes = {
  initialItems: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      quantity: PropTypes.number.isRequired
    })
  )
};

CommonHooks.defaultProps = {
  initialItems: [
    { id: 1, name: 'Item 1', quantity: 1 },
    { id: 2, name: 'Item 2', quantity: 2 },
  ]
};
