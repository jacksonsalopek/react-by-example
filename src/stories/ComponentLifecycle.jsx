import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'

/**
 * Preformatted text element to display in the log.
 */
const TextLogItem = styled.pre`
  font-family: monospace;
  font-size: 12px;
  color: ${props => props.color};
`;

/**
 * Illustrates the lifecycle of a component. 
 * 
 * <b>DEV NOTE</b>: Feel free to change any code while this page is open, it will
 * show you the lifecycle of the component as it updates.
 */
export const ComponentLifecycle = ({ textForStage, ...props }) => {
  const [stage, setStage] = React.useState(undefined);
  const [textLog, setTextLog] = React.useState([]);

  const logAndSetStage = (stage) => {
    setStage(stage);
    setTextLog([...textLog, textForStage[stage]]);
  };

  const getColorForText = (text) => {
    if (text === textForStage['mount']) {
      return 'green';
    } else if (text === textForStage['update']) {
      return 'blue';
    } else if (text === textForStage['unmount']) {
      return 'red';
    }
    return 'black';
  };

  const logElements = textLog.map((text, index) => 
    <TextLogItem
      key={`cl-${index}`}
      color={getColorForText(text)}
      className='text-sm leading-6 flex ligatures-none overflow-auto'
    ><code>{'>'}&nbsp;</code>{text}</TextLogItem>
  );

  React.useEffect(() => {
    logAndSetStage('mount');
  }, []);

  React.useEffect(() => {
    if (stage === 'mount') {
      logAndSetStage('update');
    }
    return () => {
      console.log(stage)
      setStage('unmount');
      if (stage === 'unmount') {
        setTextLog([...textLog, textForStage[stage]]);
        alert(textForStage[stage])
      }
    };
  }, [stage]);

  return (
    <div>
      {logElements}
    </div>
  );
};

ComponentLifecycle.propTypes = {
  /** Text to display at each point of the component lifecycle */
  textForStage: PropTypes.object,
};

ComponentLifecycle.defaultProps = {
  textForStage: {
    'mount': 'Component mounted',
    'update': 'Component updated',
    'unmount': 'Component unmounted',
  },
};
