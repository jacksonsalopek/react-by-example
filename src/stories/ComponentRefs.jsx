import React from 'react';

export const ComponentRefs = ({ ...props }) => {
  const childRef = React.useRef(null);
  const otherChildRef = React.useRef(null);
  const setRef = (element) => {
    childRef.current = element;
  }

  React.useEffect(() => {
    childRef.current.innerText = 'I was mutated by a ref callback!';
    otherChildRef.current.innerText = 'I was also mutated, but by a ref const!';
  }, [])

  return (
    <>
      <div ref={setRef} />
      <div ref={otherChildRef} />
    </>
  )
};

ComponentRefs.propTypes = {};

ComponentRefs.defaultProps = {};
