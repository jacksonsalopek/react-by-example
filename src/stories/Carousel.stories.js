import { Carousel } from "./Carousel";

export default {
	title: "Example/Carousel",
	component: Carousel,
	tags: ["autodocs"],
};

export const Default = {};
