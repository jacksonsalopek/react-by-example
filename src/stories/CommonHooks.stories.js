import { CommonHooks } from "./CommonHooks";

export default {
	title: "Example/CommonHooks",
	component: CommonHooks,
	tags: ["autodocs"],
};

export const Default = {};

export const WithNoItems = {
	args: {
		initialItems: [],
	},
};

export const WithLotsOfItems = {
	args: {
		initialItems: Array(100)
			.fill({})
			.map((_, i) => {
				return {
					id: i,
					name: `Item ${i}`,
					quantity: i,
				};
			}),
	},
};
