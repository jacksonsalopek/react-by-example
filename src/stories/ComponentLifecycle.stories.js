import { ComponentLifecycle } from "./ComponentLifecycle";

export default {
	title: "Example/ComponentLifecycle",
	component: ComponentLifecycle,
	tags: ["autodocs"],
};

export const Default = {};
