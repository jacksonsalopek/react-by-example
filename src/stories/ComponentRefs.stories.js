import { ComponentRefs } from "./ComponentRefs";

export default {
	title: "Example/ComponentRefs",
	component: ComponentRefs,
	tags: ["autodocs"],
};

export const Default = {};
