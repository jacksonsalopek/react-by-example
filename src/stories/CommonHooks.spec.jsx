import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import * as stories from "./CommonHooks.stories";

const { Default, WithNoItems, WithLotsOfItems } = composeStories(stories);

describe("CommonHooks", () => {
	it("should match snapshot (default)", () => {
		render(<Default />);
		expect(screen.getByTestId("common-hooks")).toMatchSnapshot();
	});

	it("should match snapshot (with no items)", () => {
		render(<WithNoItems />);
		expect(screen.getByTestId("common-hooks")).toMatchSnapshot();
	});

	it("should match snapshot (with lots of items)", () => {
		render(<WithLotsOfItems />);
		expect(screen.getByTestId("common-hooks")).toMatchSnapshot();
	});
});
