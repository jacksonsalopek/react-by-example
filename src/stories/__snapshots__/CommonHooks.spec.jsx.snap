// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CommonHooks > should match snapshot (default) 1`] = `
<div
  data-testid="common-hooks"
>
  <div>
    Item Total: 
    3
  </div>
  <button
    class="sc-beqWaB idsqWt"
  >
    Add Item
  </button>
  <ul>
    <li>
      Item 1
       - 
      1
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 2
       - 
      2
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
  </ul>
</div>
`;

exports[`CommonHooks > should match snapshot (with lots of items) 1`] = `
<div
  data-testid="common-hooks"
>
  <div>
    Item Total: 
    4950
  </div>
  <button
    class="sc-beqWaB idsqWt"
  >
    Add Item
  </button>
  <ul>
    <li>
      Item 0
       - 
      0
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 1
       - 
      1
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 2
       - 
      2
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 3
       - 
      3
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 4
       - 
      4
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 5
       - 
      5
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 6
       - 
      6
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 7
       - 
      7
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 8
       - 
      8
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 9
       - 
      9
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 10
       - 
      10
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 11
       - 
      11
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 12
       - 
      12
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 13
       - 
      13
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 14
       - 
      14
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 15
       - 
      15
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 16
       - 
      16
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 17
       - 
      17
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 18
       - 
      18
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 19
       - 
      19
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 20
       - 
      20
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 21
       - 
      21
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 22
       - 
      22
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 23
       - 
      23
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 24
       - 
      24
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 25
       - 
      25
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 26
       - 
      26
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 27
       - 
      27
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 28
       - 
      28
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 29
       - 
      29
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 30
       - 
      30
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 31
       - 
      31
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 32
       - 
      32
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 33
       - 
      33
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 34
       - 
      34
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 35
       - 
      35
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 36
       - 
      36
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 37
       - 
      37
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 38
       - 
      38
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 39
       - 
      39
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 40
       - 
      40
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 41
       - 
      41
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 42
       - 
      42
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 43
       - 
      43
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 44
       - 
      44
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 45
       - 
      45
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 46
       - 
      46
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 47
       - 
      47
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 48
       - 
      48
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 49
       - 
      49
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 50
       - 
      50
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 51
       - 
      51
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 52
       - 
      52
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 53
       - 
      53
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 54
       - 
      54
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 55
       - 
      55
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 56
       - 
      56
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 57
       - 
      57
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 58
       - 
      58
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 59
       - 
      59
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 60
       - 
      60
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 61
       - 
      61
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 62
       - 
      62
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 63
       - 
      63
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 64
       - 
      64
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 65
       - 
      65
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 66
       - 
      66
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 67
       - 
      67
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 68
       - 
      68
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 69
       - 
      69
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 70
       - 
      70
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 71
       - 
      71
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 72
       - 
      72
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 73
       - 
      73
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 74
       - 
      74
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 75
       - 
      75
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 76
       - 
      76
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 77
       - 
      77
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 78
       - 
      78
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 79
       - 
      79
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 80
       - 
      80
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 81
       - 
      81
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 82
       - 
      82
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 83
       - 
      83
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 84
       - 
      84
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 85
       - 
      85
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 86
       - 
      86
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 87
       - 
      87
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 88
       - 
      88
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 89
       - 
      89
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 90
       - 
      90
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 91
       - 
      91
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 92
       - 
      92
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 93
       - 
      93
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 94
       - 
      94
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 95
       - 
      95
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 96
       - 
      96
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 97
       - 
      97
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 98
       - 
      98
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
    <li>
      Item 99
       - 
      99
      <button
        class="sc-beqWaB fRJOXc"
        color="red"
      >
        Remove
      </button>
    </li>
  </ul>
</div>
`;

exports[`CommonHooks > should match snapshot (with no items) 1`] = `
<div
  data-testid="common-hooks"
>
  <div>
    Item Total: 
    0
  </div>
  <button
    class="sc-beqWaB idsqWt"
  >
    Add Item
  </button>
  <ul />
</div>
`;
